library/core/src/num/nonzero.rs

```diff
@@ -90,6 +90,26 @@ impl_zeroable_primitive!(
 ///
 /// assert_eq!(size_of::<Option<NonZero<u32>>>(), size_of::<u32>());
 /// ```
+///
+/// # Layout
+///
+/// `NonZero<T>` is guaranteed to have the same layout and bit validity as `T`
+/// with the exception that the all-zero bit pattern is not a valid instance.
```

Comment:
My mind wants to ask "instance of what?" when reading this sentence. How do you feel about
```suggestion
/// with the exception that the all-zero bit pattern is invalid.
```
?